src/sdk/model/mixins/entity-with-perf-samples.ts

```diff
@@ -0,0 +1,31 @@
+import { PerfSamplesRequestJson } from '../json/perf-samples-request';
+import { Iland } from '../../iland';
+import { PerfCounterJson, PerfSampleSerieJson } from '../json/perf-samples';
+
+export class EntityWithPerfSamples {
+  uuid: string;
+  apiPrefix: string;
+
+  async getPerfCounters(): Promise<Array<PerfCounterJson>> {
+    return Iland.getHttp().get(`${this.apiPrefix}/${this.uuid}/performance-counters`).then((response) => {
+      return response.data as Array<PerfCounterJson>;
+    });
+  }
+
+  async getPerfSamples(request: PerfSamplesRequestJson): Promise<PerfSampleSerieJson> {
```

Comment:
One last thing on these @fradinni . You shouldn't be returning the Json interface responses at this layer, they should be wrapped by corresponding classes. We're doing this so that we have a way to provide convenience methods associated with the data, to convert Json types to more useful javascript types, and to provide a buffer between server response format and the client integration code so that we can have better control over when we need to break compatibility with future changes.